### DISCUSSION TITLE
Compilation error observed on upgrading open-iscsi from version 2.0.873 to 2.1.9

### DISCUSSION THREAD

**Discussion by Tuhsi** [open-iscsi/open-iscsi#446] [Q&A]
Hi,
I am trying to upgrade open-iscsi from version 2.0.873 to 2.1.9 with 2.6.38 kernel
Trying to compile the latest version of open-iscsi 2.1.9. Getting the following libkmod.h error.
iface.c:36:10: fatal error: libkmod.h: No such file or directory
#include <libkmod.h>
^~~~~~~~~~~
compilation terminated.
make: *** [<builtin>: iface.o] Error 1

Could you please suggest which version of open-iscsi is compatible with 2.6.38 kernel version ?
or is there any way to disable need for libkmod in latest open-iscsi version?

**Comment by cleech** [comment]
There are some library dependencies that have been added in the decade since 2.0.873.
But, you're also running a kernel version from 2011?
What are you hoping to gain by upgrading the Open-iSCSI userspace with such an old kernel?

**Reply by Tuhsi** [reply]
Hi,

As the open-iscsi version 2.0.873 reportedly vulnerable with few CVEs, Idea is try upgrade to latest open-iscsi version which is not having any open vulnerabilities. 

To use the latest version, do we have any prerequisite? Can it be used only with specific kernel version or above?

could you please list the library dependencies for the latest open-iscsi version?

Thanks

**Reply by cleech** [reply]
> As the open-iscsi version 2.0.873 reportedly vulnerable with few CVEs, Idea is try upgrade to latest open-iscsi version which is not having any open vulnerabilities.

Doing a CVE search, the only one I see newer than 2.0.873 is CVE-2017-17840.
FYI, that's in the iscsiuio component which should only be in use if using bnx2i or qedi offload hardware.

> To use the latest version, do we have any prerequisite? Can it be used only with specific kernel version or above?
> 
> could you please list the library dependencies for the latest open-iscsi version?

libkmod was made a hard dep and the alternative code to fork/exec modprobe was removed in 2018.
libcrypto from OpenSSL became a requirement for the CHAP code in 2019.
libmount from util-linux was used for the "safe session logout" check in 2015.
libsystemd is an optional dependency

**Reply by Tuhsi** [reply]
Hi,

Thanks for your response. I see two more CVEs reported with version 2.0.873
CVE-2020-13987
CVE-2020-17437
I believe these CVEs are resolved in open-iscsi version 2.1.3. 

I am trying to bring in the dependent libraries(which you listed) for successful compilation of open-iscsi version 2.1.9.
I getting below errors related to libisns
**_**discovery.c:54:10: fatal error: libisns/isns.h: No such file or directory
#include <libisns/isns.h>
^~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [<builtin>: discovery.o] Error 1**_**

Looks like libisns code is also removed from latest open-iscsi. 
Do you suggest anything to get away from this error or can I use the open-isns code from older version?

With respect to the libkmod error, In case I could not use libkmod, can I reuse the code to fork/exec modprobe instead of libkmod? do you see any compatibility issue of using this old method with latest open-iscsi version?


Thanks

**Comment by gonzoleeman** [comment]
On Apr 11, 2024, at 1:44 PM, Tuhsi ***@***.***> wrote:
> 
> 
> Hi,
> 
> Thanks for your response. I see two more CVEs reported with version 2.0.873
> CVE-2020-13987 <https://github.com/advisories/GHSA-g438-vfc9-cq65>
> CVE-2020-17437 <https://github.com/advisories/GHSA-cfgh-w4j7-hfhp>
> I believe these CVEs are resolved in open-iscsi version 2.1.3.
> 

I’m pretty sure that, as Chris said, those are in iscsiuio.
> I am trying to bring in the dependent libraries(which you listed) for successful compilation of open-iscsi version 2.1.9.
> I getting below errors related to libisns
> discovery.c:54:10: fatal error: libisns/isns.h: No such file or directory
> #include <libisns/isns.h>
> ^~~~~~~~~~~~~~~~
> compilation terminated.
> make[3]: *** [: discovery.o] Error 1
> 
> Looks like libisns code is also removed from latest open-iscsi.
> Do you suggest anything to get away from this error or can I use the open-isns code from older version?
> 
> With respect to the libkmod error, In case I could not use libkmod, can I reuse the code to fork/exec modprobe instead of libkmod? do you see any compatibility issue of using this old method with latest open-iscsi version?
> 
> Thanks
> 

Open-isns is optional in the build now for open-iscsi. So you can build it without open-isns if you wish. Since iSNS isn’t used much as far as I know, I made it optional.

You can also just load the open-isns development package. Or you get get open-isns from github and install it yourself.See https://github.com/open-iscsi/open-isns

— 
Lee

**Comment by Tuhsi** [comment]
I don't find iscsiuio source code as part of open-iscsi version 2.0.873. 
If all those three CVEs are related to iscsiuio component, Can I safely assume that these 3 CVEs are **not applicable** to open-iscsi version 2.0.873?

Thanks

**Comment by gonzoleeman** [comment]
> I don't find iscsiuio source code as part of open-iscsi version 2.0.873. If all those three CVEs are related to iscsiuio component, Can I safely assume that these 3 CVEs are **not applicable** to open-iscsi version 2.0.873?
> 
> Thanks

Yes, that is correct. The iscsiuio directory was added for 2.0.874.

**Comment by gonzoleeman** [comment]
Is this still an issue?
